### PR TITLE
feature/connect to database

### PR DESCRIPTION
**Pull request by huytuong010101**
- Connect to mySQL
- CRUD with mySQL

### CODE REVIEW

File: src/database/init_db.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+db = require('./connection.js')
+
+create_sql = `CREATE TABLE user ( id INT NOT NULL AUTO_INCREMENT , name VARCHAR(255) NOT NULL , age INT unsigned, gender BOOLEAN , PRIMARY KEY (id))`
+seed_sql = `INSERT INTO user(name, age, gender) values ('Nguyen Van A', 18, true), ('Nguyen Van B', 20, false), ('Tran Thi C', 10, true)`
+
+db.query(create_sql, (err, result) => {
```
Comment (dangphu2412) [inline src/database/init_db.js]: variable missing const

File: src/database/init_db.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        console.log(err)
+    } else {
+        db.query(seed_sql, (err, result) => {
+            if (err){
+                console.log(err);
+            } else {
+                console.log("Success init database...")
+            }
+        })
+    }
```
Comment (dangphu2412) [inline src/database/init_db.js]:
```
if (err) {
   console.log(err);
   return;
}

db.query(...)
```

File: src/users/router.js [hunk 3]
```diff
@@ -12,78 +12,109 @@ const allUsers = [
 router.get("/", function (req, res, next) {
   const nameQuery = req.query.name;
 
+  // Search by name
   if (nameQuery) {
-    const filteredUser = allUsers.filter((user) => user.name.includes(nameQuery));
-    return res.json(filteredUser);
+    // const filteredUser = allUsers.filter((user) => user.name.includes(nameQuery));
```
Comment (dangphu2412) [inline src/users/router.js]: Remove comment

File: src/users/router.js [hunk 4]
```diff
@@ -12,78 +12,109 @@ const allUsers = [
...
 });
 
 // Delete one user
-router.delete("/:id", function (req, res, next) {
+router.delete("/:id", function (req, res) {
   const userId = parseInt(req.params.id, 10);
 
-  const removedUser = allUsers.filter((user) => user.id != userId);
-  return res.json(removedUser);
+  const sql = "Delete from user where id=?"
```
Comment (dangphu2412) [inline src/users/router.js]: About query we should follow some convention. For example of postgres:https://www.postgresql.org/docs/15/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
```
DELETE from user WHERE id = ?
```

File: src/database/init_db.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const db = require('./connection.js')
+
+const create_sql = `CREATE TABLE user ( id INT NOT NULL AUTO_INCREMENT , name VARCHAR(255) NOT NULL , age INT unsigned, gender BOOLEAN , PRIMARY KEY (id))`
+const seed_sql = `INSERT INTO user(name, age, gender) values ('Nguyen Van A', 18, true), ('Nguyen Van B', 20, false), ('Tran Thi C', 10, true)`
```
Comment (dangphu2412) [inline src/database/init_db.js]: Pls convert to camelCase. For example: createSql

File: src/users/router.js (hunk 4 above)
Comment (huytuong010101) [reply src/users/router.js]: o... what wrong here bro? my sql command is same with yours :3

**Pull request merged**